### ISSUE TITLE
problem with GC

### ISSUE THREAD

**Issue by nonsense** [ethersphere/swarm#1920]
As discussed in the #core chat room, it appears that there are still bugs with localstore GC, which result in the following storage indices (`swarm-private-38` - broken ; `swarm-private-0` - correct):

```
➜  ~ echo '{"jsonrpc":"2.0","method":"bzz_storageIndices","id":1}' | websocat ws://127.0.0.1:8001/api/v1/namespaces/pushsync/pods/http:swarm-private-38:8546/proxy/ -n --one-message | jq

{
  "jsonrpc": "2.0",
  "id": 1,
  "result": {
    "gcExcludeIndex": 0,
    "gcIndex": 188484,
    "gcSize": 188484,
    "pinIndex": 0,
    "pullIndex": 1036970,
    "pushIndex": 0,
    "retrievalAccessIndex": 188484,
    "retrievalDataIndex": 1073923
  }
}
➜  ~ echo '{"jsonrpc":"2.0","method":"bzz_storageIndices","id":1}' | websocat ws://127.0.0.1:8001/api/v1/namespaces/pushsync/pods/http:swarm-private-0:8546/proxy/ -n --one-message | jq

{
  "jsonrpc": "2.0",
  "id": 1,
  "result": {
    "gcExcludeIndex": 0,
    "gcIndex": 192540,
    "gcSize": 192540,
    "pinIndex": 0,
    "pullIndex": 192540,
    "pushIndex": 0,
    "retrievalAccessIndex": 192540,
    "retrievalDataIndex": 192540
  }
}
➜  ~
```

One idea by @janos is that this is due to `handleChunkDelivery` and namely the `ModePutRequest` at:

```
	if po >= depth || peerPO < po {
		mode = chunk.ModePutSync
	} else {
		// do not sync if peer that is sending us a chunk is closer to the chunk then we are
		mode = chunk.ModePutRequest
	}
```

**Comment by janos** [comment]
I think that it is useful to have additional information about the issue.

@nonsense Could you confirm that --mode-sync=all is used for all nodes and that the upload by smoke tests happens only on node 0? It would be good to have a reference on git about which code is used for running the tests.

I am confirming that the issue is present on current master 5bfff339a2701cc285566eecbd77530e9879eb5c, but still did not find the root cause.

**Comment by nonsense** [comment]
@janos yes, `--mode-sync=all` and all uploads go to `swarm-private-0`. `storesize` is set to 200k and seems to be respected by `swarm-private-0` based on the debug information pasted above.

What other additional information do you need?

**Comment by janos** [comment]
In general it would be very helpful to include reproduction steps in the big report. I know how to trigger the issue on a new namespace, but the bug report by itself is not clear on that. It is nice that pushsync namespace still exists to inspect nodes, but that is not permanent.

One interesting thing is that `swarm-private-1` has all indexes with the same length, just like the uploading node.

**Comment by nonsense** [comment]
The issue doesn't happen on all nodes, so it is not deterministically reproducible.

Steps to reproduce - run a Swarm network with:
- N nodes, where N >= 50
- `sync-mode=all`
- low value of `storesize` for example 200k, so that GC gets triggered after a few uploads
- cronjob with periodic file uploads so that you reach `storesize`

Problem: After most nodes start to trigger GC on every upload (since they have reached the `storesize`), usually there are some nodes that go well over the `storesize` limit and use much more data disk.

**Comment by acud** [comment]
should be fixed in #1940
